app/controllers/permissions_controller.rb

```diff
@@ -1,24 +1,42 @@
 class PermissionsController < ApplicationController
   include FiltersHelper
   include TaxonomyHelper
-  respond_to :js
 
-  def index
-    type = params[:resource_type].presence
-    @permissions = Permission.where(:resource_type => type)
-    @search_path = search_path(type)
-    @granular = granular?(type)
-
-    if @granular
-      resource_class = Filter.get_resource_class(type)
-      @show_organizations = show_organization_tab? && resource_class.allows_organization_filtering?
-      @show_locations = show_location_tab? && resource_class.allows_location_filtering?
-    end
+  def show_resource_types_with_translations
+    resources = Permission.resources
+    render :json => {
+      :resource_types => resources.map do |resource|
+                           resource_class = Filter.get_resource_class(resource)
+                           granular = granular?(resource)
+                           show_organizations = false
+                           show_locations = false
+                           if granular
+                             show_organizations = show_organization_tab? && resource_class.allows_organization_filtering?
+                             show_locations = show_location_tab? && resource_class.allows_location_filtering?
+                           end
+                           {
+                             :translation => _(resource_class || r),
```

Comment:
I am getting an error on:
```rb
undefined local variable or method `r' for #<PermissionsController:0x000000000442a0>
```

should be `resource` ?

Reply:
Yes, thank you